source/dem/particle_particle_fine_search.cc

```diff
@@ -63,17 +55,19 @@ ParticleParticleFineSearch<dim>::particle_particle_fine_search(
   // is the output of broad search. If a pair is in vicinity (distance <
   // threshold), it is added to the local_adjacent_particles
   for (auto const &[particle_one_id, second_particle_container] :
-       local_contact_pair_candidates)
+       container_manager.local_contact_pair_candidates)
     {
       if (!second_particle_container.empty())
         {
-          auto               particle_one = particle_container[particle_one_id];
+          auto particle_one =
+            container_manager.particle_container[particle_one_id];
```

Comment:
I would pre-fetch the container in a temporary variable (like copy the reference of it) instead of accessing the  class all the time . :)